### PR TITLE
fix(tedge-apt-plugin): remove symlink using rm -f instead of using conditional logic

### PR DESCRIPTION
**Pull request by reubenmiller**
## Proposed changes

Simplify the post remove maintainer script for `tedge-apt-plugin` to always remove the symlink.

Use `rm -f` which does not fail if the file does not exist (as this also avoid race conditions between the test and action).

The previous logic also was erroneous (as indicated by shellcheck rule [SC2107](https://www.shellcheck.net/wiki/SC2107)).

image

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

[#2113](https://github.com/thin-edge/thin-edge.io/issues/2113)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: configuration/debian/tedge-apt-plugin/postrm [hunk 1]
```diff
@@ -2,7 +2,4 @@
 
 set -e
 
-if [ ! -e /etc/tedge/sm-plugins/apt && ! -L /etc/tedge/sm-plugins/apt ];
-then
-   rm /etc/tedge/sm-plugins/apt
-fi
+rm -f /etc/tedge/sm-plugins/apt
```
Comment (PradeepKiruvale) [inline configuration/debian/tedge-apt-plugin/postrm:5]: I added those checks to make sure that the link exists. If we try to remove it without checking, doesn't the command fail?

Comment (reubenmiller) [reply configuration/debian/tedge-apt-plugin/postrm:5]: I put a comment about this in the PR description ;)

> Use rm -f which does not fail if the file does not exist (as this also avoid race conditions between the test and action).

Comment (reubenmiller) [reply configuration/debian/tedge-apt-plugin/postrm:5]: Short answer, no `rm -f` does not fail if the file does not exist due to the `-f` option.

https://linux.die.net/man/1/rm

> -f, --force
>     ignore nonexistent files, never prompt

**Review by rina23q — APPROVED** [review]
Nice improvement. Actually in the previous logic, if non symbolic link (e.g. binary) exists as `rm /etc/tedge/sm-plugins/apt` by chance, the `postrm` didn't remove it. I think it was useless to consider if the target is a link or something else. So just remove :D

File: configuration/debian/tedge-apt-plugin/postrm (hunk 1 above)
Comment (PradeepKiruvale) [reply configuration/debian/tedge-apt-plugin/postrm:5]: I got it, thanks!

**Review by PradeepKiruvale — APPROVED** [review]

**Pull request merged**